### PR TITLE
Refactor display to show the entire description

### PR DESCRIPTION
**Pull request by djmitche**
This switches from two columns with carefully-aligned heights of the
divs in those columns, to a float-based layout with two divs floated on
each row, separated by `clear: all`.  This avoids the need to balance
the number of fixed-height lines on each side, and thus allows
arbitrary-height descriptions with "normal" wrapping.  This is a big
usability improvement, as the descriptions are important content for
users!

This also adjusts the right row to always show the title first, then the
description, then the chips.

There's other things I'd like to change here, but this is the big one so I tried not to get carried away.

I copied this technique from the docker API docs.  I have only tried this on Firefox Release :)

### CODE REVIEW

File: src/components/SchemaTable/TableLayout.jsx [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+    boxSizing: 'border-box',
+    borderTop: `${theme.spacing(0.25)}px solid ${theme.palette.divider}`,
+    overflowX: 'wrap',
+  },
+  right: {
+    float: 'left',
+    width: '50%',
+    boxSizing: 'border-box',
+    borderTop: `${theme.spacing(0.25)}px solid ${theme.palette.divider}`,
+    overflowX: 'wrap',
```
Comment (helfi92) [inline src/components/SchemaTable/TableLayout.jsx:25]: hm, I don't think `wrap` is a valid css value. Maybe we should change this to `auto`? That would probably fix the issue with long regexes not being scrollable.

Screen Shot 2020-09-17 at 4 18 01 PM

After changing it to `auto`:

Screen Shot 2020-09-17 at 4 20 12 PM

This comment applies to the class right above it as well.

**Review by helfi92 — CHANGES_REQUESTED** [review]
#### Before

Screen Shot 2020-09-17 at 4 00 24 PM

Screen Shot 2020-09-17 at 4 02 34 PM

Screen Shot 2020-09-17 at 4 30 42 PM

#### After

1. Not the right text color

Screen Shot 2020-09-17 at 3 58 54 PM

2. The padding for the braces seems to have been removed.

Screen Shot 2020-09-17 at 3 58 47 PM

3. The continuation token title and description are too close to the gray line separator.

Screen Shot 2020-09-17 at 4 02 40 PM

4. Some "breathing space" between the description and the chips labels would be good

Screen Shot 2020-09-17 at 4 10 50 PM

5. Text is not vertically aligned with the left section

Screen Shot 2020-09-17 at 4 31 21 PM

File: src/components/SchemaTable/TableLayout.jsx [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+   * right-side diff */
+
+  return (
+    <div className={classes.wrapper}>
+      {rows.map(({ leftRow, rightRow }, i) => {
+        return (
+          <Fragment key={`row-${i}`}>
+            <div className={classes.left}>{leftRow}</div>
+            <div className={classes.right}>{rightRow}</div>
+            <div className={classes.break} />
```
[the remark this answers is not in the document]
Comment (djmitche) [reply src/components/SchemaTable/TableLayout.jsx:46]: I totally copied this from the Docker API docs, so no credit to me :)

**Comment by djmitche** [comment]
Is there a way to change the them in the Story tool?

**Comment by helfi92** [comment]
> Is there a way to change the them in the Story tool?

I believe there is. If you have the story tool running via `yarn start` then it should pick up any changes you do locally.

File: src/components/SchemaTable/TableLayout.jsx (hunk 2 above)
[the remark this answers is not in the document]
Comment (JiwoonKim) [reply src/components/SchemaTable/TableLayout.jsx:46]: This is great! I think it may also be able to solve the horizontal border issue addressed in #97 👍

**Comment by djmitche** [comment]
OK, I got a *little* carried away and did some refactors and upgraded Storybook to allow changing the theme globally.  But I think I've taken care of all of the spacing issues, etc.  Mostly, the problem was that I'd kind of ignored the styles.

File: .storybook/preview.js [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+                className={classes.spacingSlider}
+                value={spacing}
+                onChange={(e, v) => setSpacing(v)}
+                max={21}
+                min={1}
+              />
+            </FormControl>
+          </form>
+        </Paper>
+*/
```
Comment (helfi92) [inline .storybook/preview.js:85]: Dangling comment?

**Review by helfi92 — COMMENTED** [review]
This is so much nicer :) Just minor comments then we should be ready to merge.

1. Devtool errors

Screen Shot 2020-09-21 at 2 08 58 PM

2. Alignment is still a bit off

Screen Shot 2020-09-21 at 2 10 15 PM

3. Chip icon is hard to see

Screen Shot 2020-09-21 at 1 53 19 PM

Also thanks for adding the ability to toggle the theme. I love it.

**Comment by djmitche** [comment]
Sorry, those were all pretty silly oversights on my part.

I think the chip icon was a pre-existing condition, but I fixed it anyway ;)

File: .storybook/preview.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+import Slider from '@material-ui/core/Slider';
+import Select from '@material-ui/core/Select';
+import FormControl from '@material-ui/core/FormControl';
+import Typography from '@material-ui/core/Typography';
+import InputLabel from '@material-ui/core/InputLabel';
+import MenuItem from '@material-ui/core/MenuItem';
+import Paper from '@material-ui/core/Paper';
+import { TASKCLUSTER_THEME } from '../src/utils/theme';
+
+export const globalTypes = {
```
Comment (helfi92) [inline .storybook/preview.js:17]: `globalTypes` is unused as well as most of the imports above.

**Review by helfi92 — COMMENTED** [review]
I would have thought the following would get fixed by the chip icon but it turns out they're separate:

Screen Shot 2020-09-23 at 2 39 25 PM

Other than that, I love the new look :)

File: .storybook/preview.js (hunk 4 above)
Comment (djmitche) [reply .storybook/preview.js:17]: Hah, I guess that file's not linted!

Comment (djmitche) [reply .storybook/preview.js:17]: oh, `globalTypes` is a storybook thing (it's exported).  I don't really understand it, but it works :)

**Review by helfi92 — APPROVED** [review]
Thanks @djmitche for the fantastic work. @JiwoonKim Any interest in taking a look to see if you're happy with the changes before we press merge? :-)

**Review by JiwoonKim — COMMENTED** [review]
@helfi92 @djmitche 

#### Overall great improvements and it looks AWESOME! 🚀 

Having the descriptions automatically wrap and align lines between the left and right column was something I've wanted to accomplish but couldn't in order to satisfy the following two conditions:

1. render the table in row-by-row fashion to implement sections for dynamically opening `$ref` types
2. maintain only two horizontal scrolls, one for each of entire columns, so that each cell does not have its own horizontal scroll

Though the changes come at a cost that now each cells render their own horizontal scroll when overflowed, I do think the improvements of the descriptions UI weight more over the costs since descriptions are very important to the users :)

#### Just a few additional issues I found are the following:



1. no vertical divider between left column and right column
2. horizontal scrolls for individual cells

And BTW, the new storybook config with the theme addon is also really cool and convenient 👍

**Review by JiwoonKim — APPROVED** [review]

**Comment by djmitche** [comment]
Thanks!! It's nice to have some validation from the author 👍🏽 

> no vertical divider between the left column and right column

I _kind of_ like this better, as it makes the right side loo more like an "annotation" than an unrelated column of information.

> horizontal scrolls for individual cells

Good spot, that does kind of stink.  I'll file an issue for it.  I agree this is less critical than getting the descriptions visible.

**Pull request merged**